### PR TITLE
Add debug messages to measure generation time

### PR DESCRIPTION
**Pull request by EricMichaelSmith**
**Patch description**
With a `--log-generation-time` flag, log the amount of time to do forward passes through the encoder and decoder during generation, as well as the number of passes through the decoder. `CUDA_LAUNCH_BLOCKING` should be set for this.

**Testing steps**
```pytest -s tests/test_tga.py```

### CODE REVIEW

File: parlai/core/torch_generator_agent.py [hunk 1]
```diff
@@ -1114,15 +1117,21 @@ def _generate(
...
         incr_state = None
 
+        total_decoder_elapsed_time = 0
+        num_decoder_forwards = 0
         for _ts in range(max_ts):
             if all((b.is_done() for b in beams)):
                 # exit early if possible
                 break
 
+            decoder_start_time = time.time()
```
Comment (stephenroller) [inline parlai/core/torch_generator_agent.py:1127]: https://stackoverflow.com/questions/17579357/time-time-vs-timeit-timeit

fyi

File: parlai/core/torch_generator_agent.py [hunk 2]
```diff
@@ -1114,15 +1117,21 @@ def _generate(
...
             score, incr_state = model.decoder(decoder_input, encoder_states, incr_state)
             # only need the final hidden state to make the word prediction
             score = score[:, -1:, :]
             score = model.output(score)
```
Comment (stephenroller) [inline parlai/core/torch_generator_agent.py:1131]: we might need to do a flag anyway. this code isn't going to be deterministic. pytorch gpu execution is asynchronous and so you can easily be significantly undercounting the model effort here. I believe we have to use CUDA_LAUNCH_BLOCKING to get accurate measurements.

File: parlai/core/torch_generator_agent.py (hunk 1 above)
Comment (jxmsML) [reply parlai/core/torch_generator_agent.py:1127]: interesting to know!!!!

Comment (EricMichaelSmith) [reply parlai/core/torch_generator_agent.py]: good point, changing

File: parlai/core/torch_generator_agent.py (hunk 2 above)
Comment (EricMichaelSmith) [reply parlai/core/torch_generator_agent.py:1138]: Yes, good point - changed

**Comment by EricMichaelSmith** [comment]
Will integrate this into a standalone benchmarking script soon

**Pull request closed without merging**